src/lib/isCircularPeer.ts

```diff
@@ -0,0 +1,38 @@
+import { Index } from '../types/IndexType'
```

Comment:
Let's move all of the `isCircularPeer` code into getPeerDependenciesFromRegistry.ts` since that is the only place it is used. I prefer co-locating code unless it is used in different places.

Reply:
Haha I originally had that but when looking around it looked like one function per file so I moved it. I'll move it back

Reply:
Sorry for the confusion! It's likely that the codebase is not consistent in this regard.